util/src/main/java/tech/pegasys/pantheon/util/enode/EnodeURL.java

```diff
@@ -49,12 +50,12 @@
   private EnodeURL(
       final BytesValue nodeId,
       final InetAddress address,
-      final Integer listeningPort,
+      final OptionalInt listeningPort,
       final OptionalInt discoveryPort) {
     checkArgument(
         nodeId.size() == NODE_ID_SIZE, "Invalid node id.  Expected id of length: 64 bytes.");
     checkArgument(
-        NetworkUtility.isValidPort(listeningPort),
+        !listeningPort.isPresent() || NetworkUtility.isValidPort(listeningPort.getAsInt()),
```

Comment:
Question: Why do we need to check if the port is available for this representation of an EnodeURL? For example, what will happen if port  1234 is not available but we are only trying to instantiate a representation of another node's EnodeURL that is listening on 1234?

Reply:
This just checks that the port is a valid port number, not that the port is available.